be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/EventValidator.scala

```diff
@@ -0,0 +1,17 @@
+package ch.epfl.pop.pubsub.graph.validators
+
+import ch.epfl.pop.model.objects.{Hash, Timestamp}
+
+trait EventValidator {
+  /**
+   * Generates a hash to validate event ids
+   *
+   * @param hash 2nd hash parameter
+   * @param timestamp 3rd hash parameter
+   * @param string 4th hash parameter
+   * @return the calculated id
+   */
+  def generateValidationId(hash: Hash, timestamp: Timestamp, string: String): Hash
```

Comment:
This abstraction seems to obfuscate things more than anything, as it doesn't really reduce the amount of code written and hides the link with the hash function.... I wouldn't mind reverting to a simple `Hash.fromStrings` as it was there before.

Reply:
I thought about it for a while and hesitated putting it into the code. Looking at it once again, i would tend to agree with you ^^ Will remove it